src/pokemon_jump.c

```diff
@@ -2022,3 +2052,355 @@ static u16 sub_802C818(void)
+        if (++sprite->data[2] & 1)
+            sprite->pos2.y = 2;
+        else
+            sprite->pos2.y = -2;
+
+        sprite->data[1] = 0;
+    }
+
+    if (sprite->data[2] > 12)
+    {
+        sprite->pos2.y = 0;
+        sprite->callback = SpriteCallbackDummy;
+    }
+}
+
+void sub_802CC88(struct PokemonJump2 *arg0, int multiplayerId)
+{
+    sub_802C9BC(arg0->unk81A8[multiplayerId]);
+    arg0->unk81A8[multiplayerId]->callback = sub_802CD08;
+}
+
+void sub_802CCB0(struct PokemonJump2 *arg0)
+{
+    int i;
+    u16 numPlayers = sub_802C8AC();
+    for (i = 0; i < numPlayers; i++)
+    {
+        if (arg0->unk81A8[i]->callback == sub_802CD08)
+        {
+            arg0->unk81A8[i]->invisible = 0;
+            arg0->unk81A8[i]->callback = SpriteCallbackDummy;
+            arg0->unk81A8[i]->subpriority = 10;
+        }
+    }
+}
+
+static void sub_802CD08(struct Sprite *sprite)
+{
+    if (++sprite->data[0] > 3)
+    {
+        sprite->data[0] = 0;
+        sprite->invisible ^= 1;
+    }
+}
+
+void sub_802CD3C(struct PokemonJump2 *arg0)
+{
+    int i;
+    u16 numPlayers = sub_802C8AC();
+    for (i = 0; i < numPlayers; i++)
+        arg0->unk81A8[i]->subpriority = arg0->unk81FC[i];
+}
+
+void sub_802CD70(struct PokemonJump2 *arg0, int multiplayerId)
+{
+    sub_802C9BC(arg0->unk81A8[multiplayerId]);
+    arg0->unk81A8[multiplayerId]->callback = sub_802CDD4;
+}
+
+bool32 sub_802CD98(struct PokemonJump2 *arg0)
+{
+    int i;
+    u16 numPlayers = sub_802C8AC();
+    for (i = 0; i < numPlayers; i++)
+    {
+        if (arg0->unk81A8[i]->callback == sub_802CDD4)
+            return TRUE;
+    }
+
+    return FALSE;
+}
+
+static void sub_802CDD4(struct Sprite *sprite)
+{
+    switch (sprite->data[0])
+    {
+    case 0:
+        PlaySE(SE_JITE_PYOKO);
+        sprite->data[1] = 0;
+        sprite->data[0]++;
+        // fall through
+    case 1:
+        sprite->data[1] += 4;
+        if (sprite->data[1] > 0x7F)
+            sprite->data[1] = 0;
+
+        sprite->pos2.y = -(gSineTable[sprite->data[1]] >> 3);
+        if (sprite->data[1] == 0)
+        {
+            if (++sprite->data[2] < 2)
+                sprite->data[0] = 0;
+            else
+                sprite->callback = SpriteCallbackDummy;
+        }
+        break;
+    }
+}
+
+void sub_802CE48(struct PokemonJump2 *arg0, s16 x, s16 y, u8 multiplayerId)
+{
+    u8 spriteId = CreateSprite(&gUnknown_082FC00C, x, y, 1);
+    if (spriteId != MAX_SPRITES)
+    {
+        gSprites[spriteId].invisible = 1;
+        arg0->unk81BC[multiplayerId] = &gSprites[spriteId];
+    }
+}
+
+// void sub_802CE9C(struct PokemonJump2 *arg0)
+// {
+//     int i;
+//     int count;
+//     u8 spriteId;
+
+//     count = 0;
+//     for (i = 0; i < 4; i++)
+//     {
+//         spriteId = CreateSprite(&gUnknown_082FBEB8[i], gUnknown_082FBEA8[count], gUnknown_082FBE58[i * 10], 2);
+//         arg0->unk81D0[count] = &gSprites[spriteId];
+//         count++;
+//     }
+
+//     for (i = 0; i < 4; i++)
```

Comment:
have you tried seperating i, j and multiple count variables (assign count to a new variable before 2nd loop)?